### PR TITLE
Allow multiple Recurring Contributions in Product Switch Journey

### PR DESCRIPTION
**Pull request by andrewHEguardian**
## What does this change?

Removes a check that a user only has one Recurring Contribution during Product Switching. There is a ticket in Conversions team to allow multiple contributions to be checked out - so we should not assume this will be the case downstream.

Also fixes an issue picked up where the RC productDetail selected for switching was not being passed through in routerState (meaning MDAPI was being called on the second step), so we are now passing through the existing state to every page.

## How to test

Take out multiple RCs (under the threshold amount) and product switch one of them via the account overview - the second switch page should load and not redirect to the Account Overview. 

Cypress tests have been updated to include a check of how many times MDAPI is called, on a full switch journey when coming directly to the URL or by clicking to switch a specific contribution.

## Accessibility

-   [ ] [Tested with screen reader](https://github.com/guardian/accessibility/blob/main/people-and-technology/03-visual.md#screen-reader)
-   [ ] [Navigable with keyboard](https://github.com/guardian/accessibility/blob/main/people-and-technology/02-physical.md#keyboard)
-   [ ] [Colour contrast passed](https://github.com/guardian/accessibility/blob/main/people-and-technology/03-visual.md#contrast)
-   [ ] [The change doesn't use only colour to convey meaning](https://github.com/guardian/accessibility/blob/main/people-and-technology/03-visual.md#use-of-colour)

### CODE REVIEW

**Review by charleycampbell — APPROVED** [review]

**Comment by andrewHEguardian** [comment]
[Product Switching - Two RCs Bug](https://trello.com/c/gXd78Rka/4041-product-switching-two-rcs-bug)

File: cypress/integration/parallel-2/productSwitch.spec.ts [hunk 1]
```diff
@@ -270,6 +297,8 @@ describe('product switching', () => {
 			cy.findByText(
 				/Your first billing date is today and you will be charged £5/,
 			).should('exist');
+
+			cy.get('@mdapi_get_contribution.all').should('have.length', 1);
```
Comment (charleycampbell) [inline cypress/integration/parallel-2/productSwitch.spec.ts:301]: Is it worth adding a test where there is more than one?

**Review by charleycampbell — COMMENTED** [review]
Only one comment about adding in a test! But other than that it looks good to me 🚀

File: cypress/integration/parallel-2/productSwitch.spec.ts (hunk 1 above)
Comment (andrewHEguardian) [reply cypress/integration/parallel-2/productSwitch.spec.ts:301]: I actually meant to do this, but put it in the wrong setup

File: cypress/integration/parallel-2/productSwitch.spec.ts [hunk 2]
```diff
@@ -221,11 +225,31 @@ describe('product switching', () => {
 		featureSwitches.accountOverviewNewLayout &&
 		featureSwitches.productSwitching
 	) {
-		it('navigates to product switching page from Account Overview', () => {
+		it('successfully completes product switching page from Account Overview', () => {
```
Comment (KentGrm) [inline cypress/integration/parallel-2/productSwitch.spec.ts:228]: Can you add a test to demonstrate a single contribution and multiple contributions, selecting something other than the first.

File: client/components/mma/switch/options/SwitchOptions.tsx [hunk 3]
```diff
@@ -68,6 +71,8 @@ const fromAppHeadingCss = css`
 
 export const SwitchOptions = () => {
 	const switchContext = useContext(SwitchContext) as SwitchContextInterface;
+	const location = useLocation();
+	const routerState = location.state as SwitchRouterState;
```
[the remark this answers is not in the document]
Comment (andrewHEguardian) [reply client/components/mma/switch/options/SwitchOptions.tsx:75]: the state is being validated at each step in `SwitchContainer` I would argue, it's merely being accessed here in order to pass it through again (where SwitchContainer will look at it)

**Pull request merged**